mipdb/sqlite.py

```diff
@@ -0,0 +1,382 @@
+        finally:
+            session.close()
+        return result
+
+    def get_data_model(self, data_model_id: int, columns: List[str]) -> Any:
+        session = self.Session()
+        try:
+            query = session.query(*[getattr(DataModel, col) for col in columns]).filter(
+                DataModel.data_model_id == data_model_id
+            )
+            result = query.one_or_none()
+        finally:
+            session.close()
+        return result
+
+    def get_values(
+        self, table, columns: List[str] = None, where_conditions: Dict[str, Any] = None
+    ) -> List[dict]:
+        session = self.Session()
+        try:
+            if columns is None:
+                columns = [
+                    col.name for col in table.columns
+                ]  # Get all columns if none are specified
+            query = session.query(*[getattr(table.c, col) for col in columns])
+
+            if where_conditions:
+                for col, value in where_conditions.items():
+                    query = query.filter(getattr(table.c, col) == value)
+
+            result = query.all()
+        finally:
+            session.close()
+        return result
+
+    def get_data_models(self, columns: List[str]) -> List[dict]:
+        session = self.Session()
+        try:
+            query = session.query(*[getattr(DataModel, col) for col in columns])
+            result = query.all()
+        finally:
+            session.close()
+        return result
+
+    def get_dataset_count_by_data_model_id(self) -> List[dict]:
+        session = self.Session()
+        try:
+            query = session.query(
+                Dataset.data_model_id,
+                sql.func.count(Dataset.data_model_id).label("count"),
+            ).group_by(Dataset.data_model_id)
+            result = query.all()
+        finally:
+            session.close()
+        return result
+
+    def get_row_count(self, table: str) -> int:
+        session = self.Session()
+        try:
+            query = session.query(sql.func.count()).select_from(sql.text(table))
+            result = query.scalar()
+        finally:
+            session.close()
+        return result
+
+    def drop_table(self, table: str) -> None:
+        session = self.Session()
+        try:
+            table = sql.Table(table, metadata, autoload_with=self._executor)
+            table.drop(bind=self._executor)
+            session.commit()
+        finally:
+            session.close()
+
+    def delete_from(self, table, where_conditions: Dict[str, Any]) -> None:
+        session = self.Session()
+        try:
+            query = session.query(table)
+            if where_conditions:
+                for col, value in where_conditions.items():
+                    query = query.filter(getattr(table.c, col) == value)
+
+            query.delete(synchronize_session=False)
+            session.commit()
+        except Exception as e:
+            session.rollback()
+            raise
```

Comment:
Is this needed?

Won't this happen by default?

Reply:
No rollback happens by default only when you have autocommit=true